lib/inspec/input_registry.rb

```diff
@@ -35,11 +42,23 @@ def register_profile_alias(name, alias_name)
       @profile_aliases[name] = alias_name
     end
 
+    # Returns an Hash, name => Input that have actually been mentioned
     def list_inputs_for_profile(profile)
       inputs_by_profile[profile] = {} unless profile_known?(profile)
       inputs_by_profile[profile]
     end
 
+    # Returns an Array of input names. This includes input names
+    # that plugins may be able to fetch, but have not actually been
+    # mentioned in the control code.
+    def list_potential_input_names_for_profile(profile_name)
+      input_names = inputs_by_profile[profile_name].keys
+      plugins.each do |plugin|
+        input_names += plugin.list_inputs(profile_name)
+      end
+      input_names.uniq
```

Comment:
I agree that writing it as a map is clearer, but I think I'd like to avoid `flat_map` - this may be subjective, but I don't think that is very commonly used (I had to look it up to verify behavior and availability in ruby 2.4+), and a flatten() call is clearer.